### ISSUE TITLE
Issues with Istio sidecar with HTTPs pod

### ISSUE THREAD

**Issue by iandyh** [istio/istio#30576]
(**NOTE**: This is used to report product bugs:
  To report a security vulnerability, please visit <https://istio.io/about/security-vulnerabilities>
  To ask questions about how to use Istio, please visit <https://discuss.istio.io>)

**Bug description**
We have a pod listening on 443(HTTPS) with its own certs managed by k8s api server. Before 1.8.x, sidecar works find with the pod, after upgrading to 1.8.x, it stop working. Below is the error log:

```
![image](https://user-images.githubusercontent.com/431658/106546034-d37cf100-654d-11eb-8e95-bafd7bd3f6b4.png)

```

This is the listener config from `istioctl`:

```
0.0.0.0         15006 Trans: tls; App: TCP TLS; Addr: *:443         Cluster: inbound|443||
0.0.0.0         15006 Trans: raw_buffer; Addr: *:443                Cluster: inbound|443||
```

[ ] Docs
[ ] Installation
[ X ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
Sidecar should be transparent to the pods.

**Steps to reproduce the bug**
Having a pod listens on 443 with its own certs and have sidecar injection on, in k8s world, that could be a validation webhook. 

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)

```
client version: 1.8.2
control plane version: 1.8.1
data plane version: 1.8.1 (45 proxies), 1.7.4 (1 proxies)
```

**How was Istio installed?**

**Environment where the bug was observed (cloud vendor, OS, etc)**

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

**Comment by howardjohn** [comment]
This is fixed in 1.8.2

**Comment by iandyh** [comment]
@howardjohn  Thanks. We verified, it's fixed in 1.8.2
